geode-core/src/main/java/org/apache/geode/distributed/internal/ClusterDistributionManager.java

```diff
@@ -350,13 +350,13 @@ static ClusterDistributionManager create(InternalDistributedSystem system,
               }
             }
           }
+        } else {
+          distributionManager.addNewMember(id); // add ourselves
```

Comment:
This block of code needs to addNewMember even when the name is "". Why does it need to move it into an else-block?

If you start a Geode process without using `start locator` or `start server`, you can do so without a `name` but it's still a new member.

Reply:
In the method, _create_(ClusterDistributionManager.java) it adds itself twice(ClusterDistributionManager constructor and there again). So it increments this attribute for both _addNewMember()_ call.

Reply:
We probably need some unit tests for ClusterDistributionManager which would be closer to the actual code than JMX.

I can't sign off on this change for now though because as far as I know the value of `nodes` is correct. I still think you should convince dev-list that the value of `nodes` is incorrect or you should leave the value alone.

Reply:
Hi @kirklund ,
I send a mail to the dev list to discuss what is the right way to fix this. No response.
I wrote an RFC with adding a new attribute. The only one response, to fix this ticket instead of adding a new attribute.